### DISCUSSION TITLE
DataTable cells can not be edited with frozenColumns and cellEditMode="lazy" in PrimeFaces 12.0.4

### DISCUSSION THREAD

**Discussion by tnagasaki555** [primefaces/community#2790] [PrimeFaces] [answered]
This works in PrimraceFaces V6.1.
After upgrading to Primeface V12.0.4, I encountered this issue.

Here is the place.

`<p:dataTable id="xxx" widgetVar="xxx" var="item" value="#{xxx}"
rowKey="#{xxx}" rowIndexVar="rowIndex"
filteredValue="#{xxx}"
paginator="true" rows="100" paginatorPosition="top" paginatorTemplate="{PreviousPageLink} {NextPageLink}"
scrollable="true" cellEditMode="lazy" editable="true" editMode="cell" frozenColumns="3"
emptyMessage="#{xxx}" style="width: 100%;" styleClass="xxx" lazy="true">
`
When I remove "frozenColumns" attribute, I can edit cells and "lazy" works. But I can not use frozenColumns.
When I remove "cellEditMode" attribute, I can edit cells but "lazy" does not work.

> cellEditMode="lazy" editable="true" editMode="cell" frozenColumns="3"

I encountered the following error when I try to edit cells.
![image](https://github.com/user-attachments/assets/f45b35e9-20db-4c33-a011-370b3984d3f6)

Does anyone know how to fix this?

Thank you,

**Comment by melloware** [comment]
i would upgrade 12.0.4 is not the latest 12.X version and its probably already fixed.

**Reply by tnagasaki555** [reply, marked as answer]
Thank you for your comment.

I tested 12.10 LTS version and checked 13.0 LTS but both was not fixed.

I found a workaround.

I removed "$" in component.js which is in primefaces-12.0.4-jakarta.jar.

At line 236:
> BEFORE:  $this.cfg.frozenColumns : c);
> AFTER:      this.cfg.frozenColumns : c);

**Reply by melloware** [reply]
@tnagasaki555 you are right this is a bug.  I will get it fixed for 14.0.6 and 15.0.0

**Reply by melloware** [reply]
Ticket opened: https://github.com/primefaces/primefaces/issues/12672
